cdap-validation/bin/testing.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/python
+
+import os
+import test_helpers as helpers
+import json
+import re
+from subprocess import call
+import ambari
+import cm
+
+##### TEST FRAMEWORK FUNCTIONS #####
+
+input_json = 'input.json'
+
+# calls all other functions
+def testing(base_vars,cluster_vars):
+    if cluster_vars['verbose'] == 2: print 'start testing'
+
+    # prep input variables
+    #cluster_vars = {}
+    input = read_input(input_json)
+    cluster_vars['host'] = input['params']['uri']
+    cluster_vars['username'],cluster_vars['password']= input['params']['userpass'].split(':')
+    cluster_vars['cluster'] = input['params']['cluster_name']
+    cluster_vars['modules'] = input['params']['modules']
+
+    # check for verbosity not set
+    #if not input['params']['verbose']:
+    #    cluster_vars['verbose'] = 'default'
+    #else:
+    #    cluster_vars['verbose'] = input['params']['verbose']
+
+    if cluster_vars['verbose'] == 2:
+        for k,v in cluster_vars.iteritems():
+            print 'cluster vars: %s=%s' % (k,v)
+
+    # detect_installed_services
+    # hold on for now
+
+    # prep the url
+    find_api='/api'
+    if cluster_vars['host'].find(find_api) == -1:
+        cluster_vars['host'] += 'api/'
+    if cluster_vars['verbose'] == 2:  print 'uri=%s' % (cluster_vars['host'])
```

Comment:
should be using a Logger instead of all this manual printing based on verbosity level